src/server/detail/save_stages_controller.h

```diff
@@ -26,7 +26,7 @@ struct SaveStagesInputs {
   Service* service_;
   std::atomic_bool* is_saving_;
   util::fb2::FiberQueueThreadPool* fq_threadpool_;
-  std::shared_ptr<LastSaveInfo>* last_save_info_;
+  LastSaveInfo* last_save_info_;
```

Comment:
This seems to be protected by the below mutex, right?
If so, please add `ABSL_GUARDED_BY(save_mu_)`